recurring_contract/tests/test_recurring_contract.py

```diff
@@ -147,7 +64,7 @@ def test_generated_invoice(self):
         # To generate invoices, the contract must be "active"
         wf_service = netsvc.LocalService('workflow')
         wf_service.trg_validate(self.uid, 'recurring.contract',
-                                contract_id, 'contract_validated',
+                                contract.id, 'contract_validated',
```

Comment:
Please use v8 method : `contract.signal_workflow('contract_validated')`
`wf_service and netsvc` are not necessary anymore.
Change this in all tests.